packages/protocol/contracts/stability/BreakerBox.sol

```diff
@@ -275,9 +277,11 @@ contract BreakerBox is IBreakerBox, Initializable, UsingRegistry {
           emit ResetSuccessful(exchangeAddress, address(breaker));
         } else {
           emit ResetAttemptCriteriaFail(exchangeAddress, address(breaker));
+          return;
```

Comment:
I'm not sure I understand the exact desired logic of this functions.
If the first breaker can be reset, we'll set trading mode to bidirectional, then if the second breaker actually should not be reset, we won't do anything about it: emit an event, return, and leave the exchange in mode 0. Is this ok, or for some reason not a possible situation?

Reply:
At this point there isn't a concept of first or second breaker, there is only the breaker that was triggered for the given exchange. If we hit this codepath we know a breaker has been triggered because the exchange is not in the default mode. The purpose of this particular block is to try and reset the breaker for this exchange. 

> If the ~~first~~ breaker can be reset, we'll set trading mode to bidirectional

This is correct. But also there is no return so we can evaluate the other breakers in case the exchange is in a state that would trigger a breaker

> then if the ~~second~~ breaker actually should not be reset, we won't do anything about it: emit an event, return, and leave the exchange in mode 0.

This is half true. If the breaker should not be reset we do not do anything, emit and return. But in this codepath it's not possible for the trading mode to be 0(default). [Line 265](https://github.com/celo-org/celo-monorepo/blob/06e2de31bf43dfabce4087a35d3d95d60dc27035/packages/protocol/contracts/stability/BreakerBox.sol#L265) above checks this. 

So the desired logic is:
- Check if a breaker has already been tripped for this exchange & try to reset if it should be
- Check if a breaker should be tripped for this exchange & update its trading mode if necessary

Maybe breaking the function into separate smaller functions may help.